### ISSUE TITLE
Azure FunctionInvoker empty Mono causes NoSuchElementException

### ISSUE THREAD

**Issue by Munken** [spring-cloud/spring-cloud-function#886]
**Describe the bug**
When I combine a `FunctionInvoker` with a `Function<Mono<X>, Mono<Y>>` that returns an empty Mono, 
then I get an exception:
```
[2022-06-23T18:19:50.521Z] Caused by: java.util.NoSuchElementException: No value present
[2022-06-23T18:19:50.522Z]      at java.base/java.util.Optional.get(Optional.java:143)
[2022-06-23T18:19:50.522Z]      at org.springframework.cloud.function.adapter.azure.FunctionInvoker.postProcessMonoFunctionResult(FunctionInvoker.java:200)
[2022-06-23T18:19:50.523Z]      at org.springframework.cloud.function.adapter.azure.FunctionInvoker.postProcessReactiveFunctionResult(FunctionInvoker.java:177)
[2022-06-23T18:19:50.523Z]      at org.springframework.cloud.function.adapter.azure.FunctionInvoker.handleRequest(FunctionInvoker.java:132)
```

It looks like it is causes by uncontionally getting the result of an optional:
```
protected O postProcessMonoFunctionResult(I rawInputs, Object functionInputs, Mono<?> functionResult,
		FunctionInvocationWrapper function, ExecutionContext executionContext
	) {
		return (O) this.convertOutputIfNecessary(rawInputs, functionResult.blockOptional().get());
	}
```

How does not specify 404 ?

**Sample**

```
public class AnriskGetByRegRiskHandler extends FunctionInvoker<RegnrRisknr, AndenRisiko> {

    @FunctionName("andenRisikoByRegRisk")
    public HttpResponseMessage execute(
            @HttpTrigger(name = "anriskByRef",
                    methods = {HttpMethod.GET},
                    route = "andenrisiko/regnr/{regnr}/risknr/{risknr}",
                    authLevel = AuthorizationLevel.ANONYMOUS)
            HttpRequestMessage<Void> request,
            @BindingName("regnr") Integer regnr,
            @BindingName("risknr") Long risknr,
            ExecutionContext context) {

        final var payload = new RegnrRisknr(regnr, risknr);

        return request
                .createResponseBuilder(HttpStatus.OK)
                .body(handleRequest(payload, context))
                .header("Content-Type", "application/json")
                .build();
    }
}

@Component("andenRisikoByRegRisk")
public class AnriskGetByRegRiskFunction implements Function<Mono<Message<RegnrRisknr>>, Mono<AndenRisiko>> {

    private final AndenRisikoRepository repository;

    public AnriskGetByRegRiskFunction(AndenRisikoRepository repository) {
        this.repository = repository;
    }

    @Override
    public Mono<AndenRisiko> apply(Mono<Message<RegnrRisknr>> mono) {
        return mono.flatMap(message -> {
            ExecutionContext targetContext = (ExecutionContext) message.getHeaders().get("executionContext");
            Logger logger = targetContext.getLogger();

            final var payload = message.getPayload();

            logger.info("Get by " + payload);

            return repository.findAndenRisikoByRegnrAndRisknr(payload.getRegnr(), payload.getRisknr());
        });
    }
}
```

**Comment by olegz** [comment]
What version are you using?

**Comment by olegz** [comment]
It appears that the code your are referring was fixed with #757 and you are not using the latest released version which is 3.2.6. So please upgrade and feel free to re-open or rase a new issue if problem persists.
